### PR TITLE
fix where backward to use scalar 0

### PR DESCRIPTION
**Pull request by ngimel**
Per title

### CODE REVIEW

**Review by Chillee — APPROVED** [review]
If it doesn't subtly break something else, then LGTM :)

**Comment by Chillee** [comment]
For functorch fails, I think we just need to add "where" decomposition here: https://github.com/pytorch/pytorch/blob/master/functorch/functorch/csrc/BatchRulesDecompositions.cpp#L220

**Comment by ngimel** [comment]
@zou3519 @chillee any advice on vmap failures here?

File: functorch/functorch/csrc/BatchRulesDecompositions.cpp [hunk 1]
```diff
@@ -233,6 +233,7 @@ TORCH_LIBRARY_IMPL(aten, FT_BATCHED_KEY, m) {
   OP_DECOMPOSE2(vsplit, int);
   OP_DECOMPOSE2(vsplit, array);
   OP_DECOMPOSE(vstack);
+  OP_DECOMPOSE(where);
```
Comment (Chillee) [inline functorch/functorch/csrc/BatchRulesDecompositions.cpp:236]: @ngimel I think you need to decompose the specific overloads. So... looks like

```suggestion
  OP_DECOMPOSE2(where, ScalarOther);
  OP_DECOMPOSE2(where, ScalarSelf);
```

**Pull request closed without merging**